### PR TITLE
Fix stackoverflow for scala traits

### PR DESCRIPTION
**Pull request by tianyu**

### CODE REVIEW

File: kotlin-plugin/src/main/kotlin/me/shika/generation/utils.kt [hunk 1]
```diff
@@ -40,7 +40,7 @@ fun IrClass.needSerializableFix(): Boolean {
 
 fun IrClass.isSerializable(): Boolean {
     return allSuperInterfaces()
-        .any { it.fqNameWhenAvailable == SERIALIZABLE_FQ_NAME || it.isSerializable() }
+        .any { it.fqNameWhenAvailable == SERIALIZABLE_FQ_NAME || it !== this && it.isSerializable() }
```
Comment (tianyu) [inline kotlin-plugin/src/main/kotlin/me/shika/generation/utils.kt:43]: I'm not sure why, but scala traits have themselves as a superinterface. This leads to StackOverflow errors when the trait does not implement `Serializable`.

File: integration-test/src/test/kotlin/DeserializationTest.kt [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
-    private fun serializeDeserialize(instance: Serializable): Serializable {
+    @Test(expected = NotSerializableException::class)
+    fun `cannot serialize non-serializable object`() {
+        serializeDeserialize(NotSerializable)
+    }
+
+    @Test(expected = NotSerializableException::class)
+    fun `cannot serialize non-serializable object from scala`() {
+        serializeDeserialize(NotSerializableFromScala)
+    }
```
Comment (tianyu) [inline integration-test/src/test/kotlin/DeserializationTest.kt:22]: This function would throw `StackoverflowError` if not for the fix.

**Comment by tianyu** [comment]
@ShikaSD, review pretty please?

**Comment by ShikaSD** [comment]
Looks reasonable to me :)
Thanks for the update

**Comment by ShikaSD** [comment]
1.4.1 should be published now 👍

**Pull request merged**